src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleEntries = sourceString.split(';');
+
+  const stylesObj = styleEntries.reduce((prev, style) => {
+    const [key, value] = style.split(':', 2);
+
+    if (key.trim() && value.trim()) {
```

Comment:
It's great that you're checking for both key and value to be non-empty before adding them to the object. This helps in filtering out any invalid or empty style declarations.